code/modules/mob/living/carbon/xenomorph/life.dm

```diff
@@ -168,31 +169,37 @@
 	//Rollercoaster of fucking stupid because Xeno life ticks aren't synchronised properly and values reset just after being applied
 	//At least it's more efficient since only Xenos with an aura do this, instead of all Xenos
 	//Basically, we use a special tally var so we don't reset the actual aura value before making sure they're not affected
-	if(on_fire) //Burning Xenos can't emit pheromones; they get burnt up! Null it baby! Disco inferno
-		current_aura = null
-	if(current_aura && plasma_stored > 5)
+
+	if(!current_aura && !leader_current_aura) //Gotta be emitting some pheromones to actually do something
+		return
+
+	if(on_fire) //Can't output pheromones if on fire
+		return
+
+	if(current_aura) //Plasma costs are already checked beforehand
+		var/phero_range = round(6 + xeno_caste.aura_strength * 2) //Don't need to do the distance math over and over for each xeno
 		if(isxenoqueen(src) && anchored) //stationary queen's pheromone apply around the observed xeno.
 			var/mob/living/carbon/xenomorph/queen/Q = src
-			var/atom/phero_center = Q
 			if(Q.observed_xeno)
-				phero_center = Q.observed_xeno
-			var/pheromone_range = round(6 + xeno_caste.aura_strength * 2)
-			for(var/mob/living/carbon/xenomorph/Z in range(pheromone_range, phero_center)) //Goes from 8 for Queen to 16 for Ancient Queen
-				if(Z.stat != DEAD && issamexenohive(Z) && !Z.on_fire)
-					switch(current_aura)
-						if("frenzy")
-							if(xeno_caste.aura_strength > Z.frenzy_new)
-								Z.frenzy_new = xeno_caste.aura_strength
-						if("warding")
-							if(xeno_caste.aura_strength > Z.warding_new)
-								Z.warding_new = xeno_caste.aura_strength
-						if("recovery")
-							if(xeno_caste.aura_strength > Z.recovery_new)
-								Z.recovery_new = xeno_caste.aura_strength
+				//The reason why we don't check the hive of observed_xeno is just in case the watched xeno isn't of the same hive for whatever reason
+				for(var/xenos in Q.hive.get_all_xenos())
+					var/mob/living/carbon/xenomorph/Z = xenos
+					if(get_dist(Q.observed_xeno, Z) <= phero_range && (Q.observed_xeno.z == Z.z) && !Z.on_fire) //We don't need to check to see if it's dead or if it's the same hive as the list we're pulling from only contain alive xenos of the same hive
```

Comment:
get_dist() works on a square shaped radius but range() does a circular one, so you need to properly calculate the distance

Reply:
Byond docs diagram seems to indicate that it's more of a square (assuming range() fully incorporates view() which has the diagram) so this should be okay